src/PostgREST/AppState.hs

```diff
@@ -476,3 +477,23 @@ checkIsFatal(SQL.SessionUsageError (SQL.QueryError _ _ (SQL.ResultError serverEr
         -> Just "Hint: Connection poolers in statement mode are not supported."
       _ -> Nothing
 checkIsFatal _ = Nothing
+
+waitForNotifications_ :: (ByteString -> ByteString -> IO()) -- ^ Callback function to handle incoming notifications
+                     -> Connection -- ^ Connection where we will listen to
+                     -> IO ()
+waitForNotifications_ sendNotification con =
+  withLibPQConnection con $ void . forever . pqFetch
+  where
+    pqFetch pqCon = do
+      mNotification <- PQ.notifies pqCon
+      case mNotification of
+        Nothing -> do
+          mfd <- PQ.socket pqCon
+          case mfd of
+            Nothing  -> panic "Error checking for PostgreSQL notifications"
+            Just _ -> do
+              -- void $ threadWaitRead fd
+              void $ threadDelay 5000000
```

Comment:
I modified `waitForNotifications` from the Hasql library and it looks like this `threadWaitRead` is not working on Windows. Changing to `threadDelay` works, for instance, and reloads the schema without problems. This limitation for Windows systems is documented in the [function definition](https://hackage.haskell.org/package/base-4.16.4.0/docs/src/Control.Concurrent.html#threadWaitRead) and specifies that builds must use `-thread`, which is already added: https://github.com/PostgREST/postgrest/blob/32b77cae6c4227216b193f6834304e90de3bd7c3/postgrest.cabal#L161).

Not sure where to go from here, maybe the problem is with `stack` or somewhere in the build process.

Reply:
@laurenceisla Nice catch :fire:! So if `threadDelay` works how about changing `threadWaitRead` conditionally (`#if defined(mingw32_HOST_OS)..`) on a PR to https://github.com/diogob/hasql-notifications/?

Reply:
Yes, `threadDelay` works but I'm not sure if it's the correct solution here. Instead of waiting for input, it verifies every quantity of seconds (5s in my example). I only used it to test if the problem was indeed the `threadWaitRead`.

Reply:
> This limitation for Windows systems is documented in the [function definition](https://hackage.haskell.org/package/base-4.16.4.0/docs/src/Control.Concurrent.html#threadWaitRead) and specifies that builds must use -threaded, which is already added:

hasql-notifications has threaded too https://github.com/diogob/hasql-notifications/blob/master/hasql-notifications.cabal#L34

Reply:
Maybe report it on hasql-notifications anyway, Diogo might be able to help.

Reply:
I have just setup my brand new Windows development environment and I verified that hasql-notifications' tests are green. I'm going now to build PostgREST and try to reproduce the issue there.

But regardless of the investigation, I can apply the `threadDelay` fix there so we wouldn't need to merge anything here.

I'll continue posting updates on this investigation on the issue https://github.com/diogob/hasql-notifications/issues/17